### ISSUE TITLE
Feature request: Monte Carlo optimisation with custom distributions

### ISSUE THREAD

**Issue by sabirjana** [robertmartin8/PyPortfolioOpt#151]
Hello Robert,
I see "Monte Carlo optimization with custom distributions" in your road map. Definitely it will make this library highly flexible if there is an option to run Monte Carlo simulations for various kinds of optimizers offered by this library with added features to specify the kind of distribution. Please confirm my understanding of the road map you mentioned and if there is any update on progress towards this.
Thanks
Sabir Jana

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by robertmartin8** [comment]
Hi Sabir,

Appreciate the suggestion! I definitely need to do some deeper research into MC portfolio optimisation. The last time I read through, it seemed like people were just generating many random portfolios then selecting the best one out of that sample, which didn't seem like it was worth implementing. 

If anyone has any input on what features a monte Carlo optimiser should offer, I'm all ears!

Best,
Robert

**Comment by naltang** [comment]
Yes, the MC method may not look smart, but still worth implementing as a last resort when all other methods fail.

**Comment by AdamTheDestroyer** [comment]
Yes, it is THE possibility for Big Portfolios like I have (up to 7000 assets).

Greetings Adam

**Comment by sidd1996** [comment]
@robertmartin8  @AdamTheDestroyer  I will be happy to contribute to this open item,do let me know .
